addons/mail/static/src/js/services/mail_document_thread_manager.js

```diff
@@ -21,17 +21,23 @@ MailManager.include({
 
     start: function () {
         this._super.apply(this, arguments);
-
+        this._documentThreadWindowStates = {};
+        this._tabId = this.call('bus_service', 'getTabId');
         // retrieve the open DocumentThreads from the localStorage
-        var state = this.call('local_storage', 'getItem', this.DOCUMENT_THREAD_STATE_KEY);
-        if (!state) {
-            this.call('local_storage', 'setItem', this.DOCUMENT_THREAD_STATE_KEY, {});
-        } else {
-            this.isReady().then(this._updateDocumentThreadWindows.bind(this, state));
+        var localStorageLength = this.call('local_storage', 'length');
+        for (var i = 0; i < localStorageLength; i++) {
+            var key = this.call('local_storage', 'key', i);
+            if (key.indexOf(this.DOCUMENT_THREAD_STATE_KEY) !== -1) {
```

Comment:
Couldn't be nice to do as follow ? The key is supposed to actually starts with the thread state key and if it's not the case (a space or something else before the key I don't know), the substring below would yield bad results.
```suggestion
            if (key.indexOf(this.DOCUMENT_THREAD_STATE_KEY) === 0) {
```